### DISCUSSION TITLE
可以实现类似vega-lite中mark为trail的大小不断变化的线条吗？如图

### DISCUSSION THREAD

**Discussion by ChristmasFox** [antvis/G2#5124] [Q&A] [answered]
绘制大小不断变化的线条以反映基础数据
[https://vega.github.io/editor/#/edited](https://vega.github.io/editor/#/edited)
<img width="340" alt="image" src="https://github.com/antvis/G2/assets/68337739/961164ba-32ee-453c-9abc-18bd644d94b0">

**Comment by ChristmasFox** [comment]
<img width="1007" alt="image" src="https://github.com/antvis/G2/assets/68337739/1d4c283a-90fc-4654-b050-967331cc7ce3">

**Reply by pearmini** [reply]
目前还不支持，我这周 V5 支持一下。

**Reply by pearmini** [reply, marked as answer]
这个 PR 支持这个功能：https://github.com/antvis/G2/pull/5137

```js
chart
  .line()
  .data({
    type: 'fetch',
    value:
      'https://gw.alipayobjects.com/os/bmw-prod/cb99c4ab-e0a3-4c76-9586-fe7fa2ff1a8c.csv',
  })
  .encode('x', (d) => new Date(d.date))
  .encode('y', 'price')
  .encode('color', 'symbol') 
  .encode('size', 'price') // 设置 size 
  .legend('size', false)
  .style('shape', 'trail'); // 设置 shape
```

**Reply by ChristmasFox** [reply]
感谢 太强了👍
